### DISCUSSION TITLE
Backing up backrest?

### DISCUSSION THREAD

**Discussion by jsbiff** [garethgeorge/backrest#419] [Q&A]
So this question isn't about using backrest to backup data. This is about how to backup the config of backrest itself, for restoration in the future? It would be nice not to have to manually re-create the web login user, setup the repo, setup the backup set, etc.

I see in the readme there are two env variables,, BACKREST_CONFIG (a file) and BACKREST_DATA (a directory).

I'm using all the defaults for backrest, so is it sufficient to copy those two things, which default to:

$HOME/.config/backrest/config.json
$HOME/.local/share/backrest

Then, when restoring: re-install backrest, start the service, stop the backrest service, and replace those two things with the backup copies, then restart the service?

**Comment by garethgeorge** [comment]
Hey, pretty interesting question! 

My recommendation would be to simply backup your backrest config. The config is expected to be self contained and contains your users, repo definitions, plans, etc. 

The BACKREST_DATA directory is where Backrest stores the operation log, the version of restic that it installs for you (which is GPG signature verified w/the restic maintainer's key), and command logs that are bundled in .tar archives. BACKREST_DATA isn't really intended to be portable or something that you backup. Instead I'd recommend relying on just restoring your config and having backrest reindex the snapshots in your repo on the machine you want to restore from. Backrest tracks most of the important metadata for restore operations in the snapshots themselves (through tags indicating the plan and backrest install that created the snapshot).

**tl;dr** Personally, I just keep a copy of my config.json in my password manager as a secure note. Backrest will never break config version compatibility outside of a major version change (and will always maintain migrations for a few versions on either side) so backing up your config manually when adding or removing repos is enough.

**Reply by zero0n3** [reply]
hi @garethgeorge 
today I find myself in the (desired) situation of having to restore my raspberry pi2.

The (fortuitous) case is the following:

I had a raspberry os lite installation where I had installed backrest and where I regularly did (as an experiment) the LAN backup of /etc/systemd and /home/pi (where the backrest configuration files were also contained).

Now I changed SD (I still have the old one as a precaution, which is why I talk about experiments) to install everything from scratch but this time using diet pi.

I reinstalled everything (to also make my own manual) but **now the dilemma**: let's say that the old SD is not accessible as today, to be able to easily recover the backrest configuration and then access the backup made with backrest (which itself contains the backrest configuration) your suggestion is therefore to keep the configuration files saved on another media in order to be able to access the json file to restore the backrest configuration and access the real backrest backup?

**Reply by garethgeorge** [reply]
Right — the easiest way to restore by far is to backup your JSON configs somewhere. When you create a new install you then copy in the JSON to load your repos and plans. I keep mine as a note in a password manager for each system I run. 

Alternatively if you don’t want to save the full JSON, you can backup your credentials for each restic repo and should be able to re-add them and import snapshots from the existing repos to a new configuration. 

The second approach works well too, but I prefer recommending users backup the JSON as that way you aren’t fiddling with credentials and env vars when you’re worrying about data loss and likely to make mistakes.

**Reply by zero0n3** [reply]
The first option will be more than fine. Thanks for the clarification

**Reply by Alk2** [reply]
It is really important to Quit Backrest and Start it again after replacing the config file from your backup.

**Reply by MissMusic** [reply]
This is how I have my disaster recovery set up:

1.1 Install backrest
1.2 Configure n+1 repos
1.3 Create one perpetual backup job per repo for Backrests own .json config. (I backup the .json daily)
1.4 Run the backup jobs once to verify they are set up correctly.
1.5 Save the backrest config json. I've added the content as a text note in my password manager (it automatically syncs a copy to all devices with it installed), you could also save it on a flash drive in your safe or whatever.

For a full DR:
2.1 Spin up a new backrest docker.
2.2 Update the .json config with the old file saved in step 1.5.
2.3 Restart Backrest to load the config
Only the initial backrest self backup jobs exist.
2.4 Download the latest backed up config .json from either of the repos.
2.5 Replace the config .json again with the latest copy.
2.6 Restart Backrest to load the config
2.7 Backrest is back with all the repos and backup jobs from the last daily backup, DR of backrest itself is complete.
2.8 Continue with DR of all other services now that you have the latest backups accessible through backrest.

The initial config .json does not need updating as long as you don't change the repos or backup jobs for backrest, and even if you do, as long as one repo and backup job is unchanged you should be able to recover.
